Fix: update GCC version used in CI builds

SlySven opened the pull request:
From 7.x to 8.x.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: update GCC version used in CI builds

From 7.x to 8.x.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: accommodate differences in GCC/Ubuntu versions

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: expression of matrix variables in yaml files

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Fix: correct a copy/paste typo

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on .github/workflows/build-mudlet.yml:
```diff
@@ -178,14 +181,18 @@ jobs:
+        # Ubuntu-18.06.4 and Ubuntu-22.04.1 do have a gcc-8 - it is the default
+        # now for the former but not the latter and for the "Ubuntu / clang"
+        # job it seems a bit suspect to have to force a gcc version given that
+        # we should not be intending to use it! Slysven 2022/12
```
I'm not certain the comment would help all that much to other developers  who come across to read it, maybe leave that in the PR description?

SlySven replied:
Done in 288394319524e282dbb717fc2fb2ff7ea561b9a7 and c5500142a8ee05b770ec16a908184bd056c82d92 ...

SlySven commented on .github/workflows/build-mudlet.yml:
```diff
@@ -178,14 +181,18 @@ jobs:
+        # Ubuntu-18.06.4 and Ubuntu-22.04.1 do have a gcc-8 - it is the default
+        # now for the former but not the latter and for the "Ubuntu / clang"
+        # job it seems a bit suspect to have to force a gcc version given that
+        # we should not be intending to use it! Slysven 2022/12
```
```suggestion
```

SlySven committed:
Update .github/workflows/build-mudlet.yml

Remove comment from code

"Ubuntu-18.06.4 and Ubuntu-22.04.1 do have a gcc-8 - it is the default now for the former but not the latter and for the "Ubuntu / clang" job it seems a bit suspect to have to force a gcc version given that we should not be intending to use it!"

Signed-off by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented on .github/workflows/codeql-analysis.yml:
```diff
@@ -119,13 +121,17 @@ jobs:
-        sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-7 7
-        sudo update-alternatives --set gcc /usr/bin/gcc-7
-        sudo update-alternatives --set g++ /usr/bin/g++-7
+        # Ubuntu-18.06.4 and Ubuntu-22.04.1 do have a gcc-8 - it is the default
+        # now for the former but not the latter and for the "Ubuntu / clang"
+        # job it seems a bit suspect to have to force a gcc version given that
+        # we should not be intending to use it! Slysven 2022/12
```
```suggestion
```

SlySven committed:
Update .github/workflows/codeql-analysis.yml

vadi2 reviewed (approved):
thx for the fix!
